sqlglot/transforms.py

```diff
@@ -914,3 +914,30 @@ def eliminate_join_marks(expression: exp.Expression) -> exp.Expression:
             where.pop()
 
     return expression
+
+
+def any_to_exists(expression: exp.Expression) -> exp.Expression:
+    """
+    Transform ANY operator to Spark's EXISTS
+
+    For example,
+        - Postgres: SELECT * FROM tbl WHERE 5 > ANY(tbl.col)
+        - Spark: SELECT * FROM tbl WHERE EXISTS(tbl.col, x -> x < 5)
+
+    Both ANY and EXISTS accept queries but currently only array expressions are supported for this
+    transformation
+    """
+    if isinstance(expression, exp.Select):
```

Comment:
Why don't we just hook into `exp.Any`'s generator for the dialects of interest?

Reply:
That was my first attempt, but I don't think it's very practical because `ANY` is a child of an `exp.Binary` node; This means that for a filter such as `WHERE <expr> <op> <any_expr>`, we will have already processed the LHS before we get to `any_sql` for the RHS. To mitigate that, we'd have to hook into `generator.py::binary()` and do the transformation there in advance.

For this reason, I thought it'd look cleaner to add a `transform` and in the future it can be extended to cover subqueries as well.